### PR TITLE
Don't update linetype when it was blank before

### LINKED ISSUE #933: Legend update shouldn't introduce new lines where linetype was blank
image

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #933

### CODE REVIEW

File: R/utilities-plotting.R [hunk 1]
```diff
@@ -440,27 +443,39 @@
...
 
-  # Extract as many properties as there are datasets from the specified config
-  # object fields.
-  if (length(pointsColor) > 1L) {
-    pointsColor <- pointsColor[1:nrow(legendCaptionData)]
+  # As many unique colors and shapes will be needed as there are groupings
+  numberOfColorsNeeded <- numberOfShapesNeeded <- nrow(legendCaptionData)
+
+  # Number of needed line types would be equal to only non-blank line types.
+  numberOfLinetypesNeeded <- nrow(legendCaptionData[legendCaptionData$linetype != "blank", ])
```
Comment (msevestre) [inline R/utilities-plotting.R:450]: don't we have an enum value for this?

Comment (IndrajeetPatil) [reply R/utilities-plotting.R]: Done.

File: R/utilities-plotting.R [hunk 2]
```diff
@@ -440,27 +443,39 @@
...
     color = pointsColor,
-    shape = pointsShape,
-    linetype = lineTypes
+    shape = pointsShape
   )
 
+  # New version of legend mappings for linetype: only non-blank line rows are replaced.
+  # Relevant only if there were any blank lines in the original legend data frame.
+  if (numberOfLinetypesNeeded > 0L) {
+    newLegendCaptionData[newLegendCaptionData$linetype != "blank", ]$linetype <- lineTypes
```
Comment (msevestre) [inline R/utilities-plotting.R:477]: Same here too.

**Comment by IndrajeetPatil** [comment]
It has nothing to do with the PR.

The plot looks different every time the test is run. This is why I am not running that test on AppVeyr. I am assuming this has something to do with how population simulations are run but not sure.

**Review by PavelBal — APPROVED** [review]

**Pull request merged**